assets/js/modules/analytics/components/settings/GA4SettingsControls.js

```diff
@@ -0,0 +1,170 @@
+import { FORM_SETUP, MODULES_ANALYTICS } from '../../datastore/constants';
+import { Select, Option } from '../../../../material-components';
+import { GA4ActivateSwitch } from '../common';
+import {
+	PropertySelect,
+	UseSnippetSwitch,
+} from '../../../analytics-4/components/common';
+const { useSelect, useDispatch } = Data;
+
+export default function GA4SettingsControls() {
+	const [ matchedProperty, setMatchedProperty ] = useState();
+	const [ matchedWebDataStream, setMatchedWebDataStream ] = useState();
+
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	// This select is needed to check whether the AdminAPI works or not.
+	useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getProperties( accountID )
+	);
+
+	const isAdminAPIWorking = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).isAdminAPIWorking()
+	);
+
+	const enableGA4 = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'enableGA4' )
+	);
+
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+
+	const {
+		matchAccountProperty,
+		matchWebDataStream,
+		setPropertyID,
+		setWebDataStreamID,
+		setMeasurementID,
+	} = useDispatch( MODULES_ANALYTICS_4 );
+
+	useUpdateEffect( () => {
+		const matchGA4Information = async () => {
+			const matchingProperty = await matchAccountProperty( accountID );
+
+			setMatchedProperty( matchingProperty );
+			if ( matchingProperty?._id ) {
+				const matchingWebDataStream = await matchWebDataStream(
+					matchingProperty._id
+				);
+				setMatchedWebDataStream( matchingWebDataStream );
+			}
+		};
+
+		if ( isAdminAPIWorking ) {
+			matchGA4Information();
+		}
+	}, [
+		accountID,
+		matchAccountProperty,
+		matchWebDataStream,
+		isAdminAPIWorking,
+	] );
+
+	const onActivate = useCallback( () => {
+		setPropertyID( matchedProperty?._id || PROPERTY_CREATE );
+		setWebDataStreamID( matchedWebDataStream?._id || WEBDATASTREAM_CREATE );
+		setMeasurementID( matchedWebDataStream?.measurementId || '' ); // eslint-disable-line sitekit/acronym-case
+	}, [
+		matchedProperty,
+		matchedWebDataStream,
+		setPropertyID,
+		setWebDataStreamID,
+		setMeasurementID,
+	] );
+
+	if ( ! isAdminAPIWorking ) {
+		return null;
+	}
+
+	const isDisabled = ! propertyID && ! enableGA4;
+
+	return (
+		<div className="googlesitekit-settings-module__fields-group">
+			<div className="googlesitekit-setup-module__inputs">
+				{ ! isDisabled && (
+					<PropertySelect
+						label={ __(
+							'Google Analytics 4 Property',
+							'google-site-kit'
+						) }
+					/>
+				) }
+				{ isDisabled && (
+					<Select
+						className="googlesitekit-analytics__select-property"
+						label={ __(
+							'Google Analytics 4 Property',
+							'google-site-kit'
+						) }
+						value={ matchedProperty?._id || '' }
+						disabled
+						enhanced
+						outlined
+					>
+						<Option value={ matchedProperty?._id || '' }>
```

Comment:
This makes it so the one value is always selected.
```suggestion
						<Option value="1">
```

Reply:
See above, I think this is correct as is. If anything, it may be even better to not even render any `Option` if there is no matching property, since in that case that `Option` is irrelevant. Still, this is more of a nitpick than a blocker. cc @eugene-manuilov